Tests/SwiftSyntaxBuilderTest/StringInterpolation.swift

```diff
@@ -93,6 +107,121 @@ final class StringInterpolationTests: XCTestCase {
       """
   }
 
+  func testInterpolationLiteralString() {
+    let a: ExprSyntax = "print(\(literal: "Hello, world!"))"
+    AssertStringsEqualWithDiff(a.description, #"print("Hello, world!")"#)
+
+    let b: ExprSyntax = "print(\(literal: "\"Hello\", world!"))"
+    AssertStringsEqualWithDiff(b.description, ##"print(#""Hello", world!"#)"##)
+
+    let c: ExprSyntax = "print(\(literal: "Hello\\#\\world!"))"
+    AssertStringsEqualWithDiff(c.description, ###"print(##"Hello\#\world!"##)"###)
+
+    let d: ExprSyntax = "print(\(literal: "Hello, world!\n"))"
+    AssertStringsEqualWithDiff(d.description, #"print("Hello, world!\n")"#)
+
+    let e: ExprSyntax = "print(\(literal: "\"Hello\", world!\n"))"
+    AssertStringsEqualWithDiff(e.description, ##"print(#""Hello", world!\#n"#)"##)
+  }
+
+  func testInterpolationLiteralInt() {
+    func test<T: ExpressibleByIntegerLiteral & ExpressibleByLiteralSyntax>(with ty: T.Type, unsigned: Bool = false) {
+      let a: ExprSyntax = "print(\(literal: 42 as T))"
+      AssertStringsEqualWithDiff(a.description, #"print(42)"#, additionalInfo: String(describing: ty))
```

Comment:
To make sure Xcode displays any test failures on the `test` line, could you pass down `line` here?

```suggestion
    func test<T: ExpressibleByIntegerLiteral & ExpressibleByLiteralSyntax>(with ty: T.Type, unsigned: Bool = false, file: StaticString = #file, line: UInt = #line) {
      let a: ExprSyntax = "print(\(literal: 42 as T))"
      AssertStringsEqualWithDiff(a.description, #"print(42)"#, additionalInfo: String(describing: ty), file: file, line: line)
```

(same for the other `AssertStringsEqualWithDiff` calls and the `test` functions in the `FloatLiteral` test)

Reply:
I actually did it this way initially, but I found that when a test case failed, I usually cared more about the value (e.g. `42`) than the type. Each of them includes the type in its `additionalInfo:` so that you can figure out exactly what failed in any case.

Reply:
If multiple tests fail I just dislike having multiple assertion failures on the same line but if you made this deliberate decision, I’m happy with it.